src/main/features/claimant-response/routes/payment-option.ts

```diff
@@ -2,25 +2,147 @@ import * as express from 'express'
+
+    if (decisionType === DecisionType.CLAIMANT || decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT) {
+      if (draft.alternatePaymentMethod.paymentOption.option.value === PaymentOption.IMMEDIATELY) {
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.IMMEDIATELY
+        courtOfferedPaymentIntention.paymentDate = MomentFactory.currentDate().add(5,'days')
+      }
+      return courtOfferedPaymentIntention
+    }
+
+    if (decisionType === DecisionType.COURT) {
+      const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
+        courtOfferedPaymentIntention.repaymentPlan = {
+          firstPaymentDate: paymentPlanFromDefendantFinancialStatement.startDate,
+          instalmentAmount: paymentPlanFromDefendantFinancialStatement.instalmentAmount,
+          paymentSchedule: Frequency.toPaymentSchedule(paymentPlanFromDefendantFinancialStatement.frequency),
+          completionDate: paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate(),
+          paymentLength: paymentPlanFromDefendantFinancialStatement.calculatePaymentLength()
+        }
+        return courtOfferedPaymentIntention
+      }
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+        const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
+        courtOfferedPaymentIntention.paymentDate = paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate()
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+
+        return courtOfferedPaymentIntention
+      }
+    }
+
+    if (decisionType === DecisionType.DEFENDANT) {
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
+        courtOfferedPaymentIntention.paymentDate = claimResponse.paymentIntention.paymentDate
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+      }
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
+        courtOfferedPaymentIntention.repaymentPlan = claimResponse.paymentIntention.repaymentPlan
+      }
+      return courtOfferedPaymentIntention
+    }
+  }
+
+  static generateCourtCalculatedPaymentIntention (draft: DraftClaimantResponse, claim: Claim, decisionType: DecisionType): PaymentIntention {
+    if (decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT && draft.alternatePaymentMethod.paymentOption.option.value === PaymentOption.BY_SPECIFIED_DATE) {
```

Comment:
From our discussion isn't the second operand superfluous? It's enough to know that it's claimant-in-favour-of-defendant

Reply:
yes, removed it now.

Reply:
raised a bug to resolve this issue in master - https://tools.hmcts.net/jira/browse/ROC-4644